bears/vcs/git/GitCommitBear.py

```diff
@@ -96,260 +47,19 @@ def get_host_from_remotes():
-            yield Result(self, 'Body of HEAD commit contains too long lines. '
-                               'Commit body lines should not exceed {} '
-                               'characters.'.format(body_line_length))
-
-    def check_issue_reference(self, body,
-                              body_close_issue: bool=False,
-                              body_close_issue_full_url: bool=False,
-                              body_close_issue_on_last_line: bool=False,
-                              body_enforce_issue_reference: bool=False):
-        """
-        Check for matching issue related references and URLs.
-
-        :param body:
-            Body of the commit message of HEAD.
-        :param body_close_issue:
-            GitHub and GitLab support auto closing issues with
-            commit messages. When enabled, this checks for matching keywords
-            in the commit body by retrieving host information from git
-            configuration. By default, if none of ``body_close_issue_full_url``
-            and ``body_close_issue_on_last_line`` are enabled, this checks for
-            presence of short references like ``closes #213``.
-            Otherwise behaves according to other chosen flags.
-            More on keywords follows.
-            [GitHub](https://help.github.com/articles/closing-issues-via-commit-messages/)
-            [GitLab](https://docs.gitlab.com/ce/user/project/issues/automatic_issue_closing.html)
-        :param body_close_issue_full_url:
-            Checks the presence of issue close reference with a full URL
-            related to some issue. Works along with ``body_close_issue``.
-        :param body_close_issue_on_last_line:
-            When enabled, checks for issue close reference presence on the
-            last line of the commit body. Works along with
-            ``body_close_issue``.
-        :param body_enforce_issue_reference:
-            Whether to enforce presence of issue reference in the body of
-            commit message.
-        """
-        if not body_close_issue:
-            return
-
-        host = self.get_host_from_remotes()
-        if host not in self.SUPPORTED_HOST_KEYWORD_REGEX:
-            return
-
-        if body_close_issue_on_last_line:
-            if body:
-                body = body.splitlines()[-1]
-            result_message = ('Body of HEAD commit does not contain any {} '
-                              'reference in the last line.')
-        else:
-            result_message = ('Body of HEAD commit does not contain any {} '
-                              'reference.')
-
+    def get_issue_regex(self, body_close_issue_full_url, host):
         if body_close_issue_full_url:
             result_info = 'full issue'
             issue_ref_regex = (
                 r'https?://{}\S+/issues/(\S+)'.format(re.escape(host)))
         else:
             result_info = 'issue'
             issue_ref_regex = r'(?:\w+/\w+)?#(\S+)'
```

Comment:
I think you can move these regexes into class constants

Reply:
I guess I will open a new issue for these then :smile:

Reply:
> I'm not sure if it's that profitable, that would be rather a performance optimization, and I think it's really negligible.
You only save a variable assignment here, the way more expensive regex compilation is not affected.

Reply:
The objective is to reduce duplication between the git & hg bears.  Ideally the hg bear uses almost all of the same moving parts as the git bear, as the git bear will be much better tested, and the hg bear will get only little use.  Using inheritance will ensure we have less overhead in maintaining hg bear.